### PR TITLE
Added warning for mispelling of componentWillReceiveProps and updated…

### PR DESCRIPTION
**Pull request by davemeetsworld**
Added a warning for the misspelling of componentWillReceiveProps

( This is my first pull request for this project )

### CODE REVIEW

**Comment by davemeetsworld** [comment]
#4039

File: docs/js/react.js [hunk 1]
```diff
@@ -5895,6 +5895,15 @@ var ReactClass = {
       ) : null);
     }
 
+    if ("production" !== "development") {
```
Comment (zpao) [inline docs/js/react.js]: Can you revert this file? We only check this in on releases.

File: src/isomorphic/modern/class/__tests__/ReactCoffeeScriptClass-test.coffee [hunk 2]
```diff
@@ -319,6 +319,24 @@ describe 'ReactCoffeeScriptClass', ->
...
+      render: ->
+        span
+          className: 'foo'
+
+    test React.createElement(NamedComponent), 'SPAN', 'foo'
+    expect(console.error.calls.length).toBe 1
+    expect(console.error.calls[0].args[0]).toBe(
+      'Warning: NamedComponent has a method called componentWillRecieveProps().
+       Did you mean componentWillReceiveProps()? The name is phrased as a
+       question because the function is expected to return a value.'
```
Comment (zpao) [inline src/isomorphic/modern/class/__tests__/ReactCoffeeScriptClass-test.coffee]: Double check this. This isn't the text of the warning.

**Comment by davemeetsworld** [comment]
Thanks for the review. I addressed the issues in the latest commit.

**Comment by zpao** [comment]
:thumbsup: Now last request: could you squash to a single commit?

**Comment by davemeetsworld** [comment]
Of course :) done.

**Comment by zpao** [comment]
thanks!

**Comment by jamischarles** [comment]
Closes https://github.com/facebook/react/issues/4039

**Comment by davemeetsworld** [comment]
cool, thank you.

**Pull request merged**